### PR TITLE
Redesign UI of the build results tab

### PR DESCRIPTION
**Pull request by saraycp**
On the redesigned request show page, we have a wide area to display the build results, not a small column as we had before.

Please, bear in mind that the code needs a lot of refactoring, but I've just done minimal changes to make it work and fit the new area. A proper refactoring will come soonish.

**How to test?**

- Make sure your user has the feature flag `request_show_page` enabled.
- Go to a request page.
- Add many repositories and architectures to the project of the package associated with the request.
- Use the refresh button to see different states of the build results.
- Play around with the question mark icon, the collapse arrow, etc.

`codeclimate` complains about the long `RequestController#show` action, we are ignoring it during the implementation of `request_show_redesign`.

~**DO NOT MERGE** until #13236 is reviewed and merged.~

### CODE REVIEW

**Comment by hennevogel** [comment]
Not counting the boxes that are always there, you have 4 levels deep of boxes now.

196802841-ef56faf1-76d5-4f77-b34a-739a88851e80

It feels very heavy and hard to parse. For a single package-repo-arch it's most evident I think

Screenshot from 2022-10-20 11-25-53

**Comment by saraycp** [comment]
I've rebased on master and added a patch to make the package page render the right partial for Build Results when the beta feature is enabled.

**Comment by saraycp** [comment]
> Not counting the boxes that are always there, you have 4 levels deep of boxes now.

BTW, I'm working on another mockup.

**Comment by saraycp** [comment]
I pushed some changes, but there is still a lot to do.

This is not the final result, but it's close. At least it replaces the design in boxes for something simpler:

Screenshot 2022-10-21 at 15-56-26 Open Build Service

**Comment by saraycp** [comment]
This PR is ready for review now, this is the current state:

build_results_mvp

File: src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml [hunk 1]
```diff
@@ -7,61 +7,48 @@
...
+                - if result.is_repository_in_db
+                  = repository_status_icon(status: result.state)
+                  %span.pl-1.d-none.d-md-inline= repository_info(result.state)
+                - else
+                  %i.fas.fa-clock.text-warning
+                  This result is outdated
+              .col-1.text-right
+                %i.fa.fa-question-circle.text-info.toggle-result-details{ title: 'Click to keep it open' }
+            .row.result-details.bg-light.mb-1.d-none
+              .col.p-3
```
Comment (dmarcoux) [inline src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:36]: This isn't a blocker, but this container and its children have a bit too much padding. We should tone this down. It can happen in another PR.

Comment (saraycp) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:36]: I agree. 

`p-3` is too much:

Screenshot 2022-10-24 at 11-27-26 Open Build Service

`p-2` is ok for me, `p-1` is not enough:

Screenshot 2022-10-24 at 11-26-55 Open Build Service

File: src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml [hunk 2]
```diff
@@ -7,61 +7,48 @@
...
+                = arch_repo_table_cell(result.repository, result.architecture, package, 'code' => result.code, 'details' => result.details)
+              .col-1.col-md-6
+                - if result.is_repository_in_db
+                  = repository_status_icon(status: result.state)
+                  %span.pl-1.d-none.d-md-inline= repository_info(result.state)
+                - else
+                  %i.fas.fa-clock.text-warning
+                  This result is outdated
+              .col-1.text-right
+                %i.fa.fa-question-circle.text-info.toggle-result-details{ title: 'Click to keep it open' }
```
Comment (hennevogel) [inline src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: This is a bit confusing now. The red parts are about the repository, the green parts are about the package.

Screenshot from 2022-10-24 11-43-05

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml (hunk 1 above)
Comment (saraycp) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml]: I've reduced the padding/margin between the children too:

Screenshot 2022-10-24 at 11-51-47 Open Build Service

File: src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml (hunk 2 above)
Comment (hennevogel) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: Suggestion: The ⚙️ should move next to 15.3 and when clicked expand the explanation (Build jobs exist for this repository) below it. The content of the grey box should move to the place there the ⚙️ is.

Comment (saraycp) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: Do you think re-ordering the content like follows could help?

small_viewport_build_results

large_viewport_build_results

This way, the order of what we display from left to right corresponds to what we display from top to bottom in the details box.

File: src/api/app/controllers/webui/package_controller.rb [hunk 3]
```diff
@@ -543,17 +543,14 @@ def buildresult
...
-                       'buildstatus'
-                     end
-
-      render partial: partial_name, locals: { buildresults: @buildresults,
-                                              index: @index,
-                                              project: @project,
-                                              collapsed_packages: params.fetch(:collapsedPackages, []),
-                                              collapsed_repositories: params.fetch(:collapsedRepositories, {}) }
+      # TODO: this is part of the temporary changes done for 'request_show_redesign'.
+      request_show_redesign_partial = 'webui/request/beta_show_tabs/build_status' if params[:in_request_show_redesign]
```
Comment (danidoni) [inline src/api/app/controllers/webui/package_controller.rb:547]:
```suggestion
      request_show_redesign_partial = 'webui/request/beta_show_tabs/build_status' if params.fetch(:inRequestShowRedesign, false)
```

File: src/api/app/assets/javascripts/webui/request_show_redesign/build_results.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ function updateBuildResultBeta() { // jshint ignore:line
   var ajaxDataShow = $('.build-results-content').data();
   // show_all comes from the buildstatus partial
   ajaxDataShow.show_all = $('#show_all').is(':checked'); // jshint ignore:line
+  ajaxDataShow.in_request_show_redesign = true; // jshint ignore:line
```
Comment (danidoni) [inline src/api/app/assets/javascripts/webui/request_show_redesign/build_results.js:19]:
```suggestion
  ajaxDataShow.inRequestShowRedesign = true;
```

File: src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml (hunk 2 above)
Comment (hennevogel) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: On small viewports you have no problem. There you have kind of a table. 

Screenshot from 2022-10-24 12-36-44

and clicking the ❓ will expand some "help" about the build/repo status.

Only on large viewports it's gets very confusing as the name of repository, it's state and the explanation of the repositories have no proximity. Same for the builds state and explanation.

File: src/api/app/controllers/webui/package_controller.rb (hunk 3 above)
Comment (saraycp) [reply src/api/app/controllers/webui/package_controller.rb]: done

File: src/api/app/assets/javascripts/webui/request_show_redesign/build_results.js (hunk 4 above)
Comment (saraycp) [reply src/api/app/assets/javascripts/webui/request_show_redesign/build_results.js]: done

File: src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml (hunk 2 above)
Comment (saraycp) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: I'm trying to find an alternative... as the repository state is coupled with the architecture... :thinking: 

Screenshot 2022-10-24 at 11-54-15 openSUSE build service Server

So far I'm moving the repository state closer to the repository name.

Comment (hennevogel) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: > repository state is coupled with the architecture

The repository state is for *all* architectures.

Comment (saraycp) [reply src/api/app/views/webui/request/beta_show_tabs/_build_status.html.haml:34]: This is obs-server right now:

Screenshot 2022-10-24 at 13-50-10 openSUSE build service Server

**Comment by hennevogel** [comment]
## L 

Screen Shot 2022-10-24 at 17 13 51

## M

Screen Shot 2022-10-24 at 17 13 59

## S

Screen Shot 2022-10-24 at 17 14 05

**Comment by hennevogel** [comment]
[Build Results Tab UI issues](https://trello.com/c/h3HQFEKH/2766-build-results-tab-ui-issues)

**Review by hennevogel — APPROVED** [review]
Reviewed by pushing commits...
